aptos-move/framework/aptos-framework/sources/event.move

```diff
@@ -54,4 +54,23 @@ module aptos_framework::event {
     public fun destroy_handle<T: drop + store>(handle: EventHandle<T>) {
         EventHandle<T> { counter: _, guid: _ } = handle;
     }
+
+    #[test_only]
+    use std::option::{Self, Option};
+
+    #[test_only]
+    public fun emitted_event<T: drop + store>(handle: &EventHandle<T>, sequence_number: u64): Option<T> {
+        let id = bcs::to_bytes(&handle.guid);
+        if (emitted_internal<T>(id, sequence_number)) {
+            option::some(emitted_event_internal<T>(id, sequence_number))
+        } else {
+            option::none()
+        }
+    }
+
+    #[test_only]
+    native fun emitted_internal<T: drop + store>(id: vector<u8>, sequence_number: u64): bool;
+
+    #[test_only]
+    native fun emitted_event_internal<T: drop + store>(id: vector<u8>, sequence_number: u64): T;
```

Comment:
> Unfortunately this is rather complicated to implement and so it was decided to split these into two functions.

Okay, it's a bit complicated, but not that hard. Here's how you return an `Option<T>` in a native function.
```
Value::struct_(Struct::pack(Value::vector_for_testing_only([event_value])))
```
Totally understand the frustration though. Hopefully one day I'll have time to implement some user-friendly value APIs for the values...

Reply:
oooh I was so close... I had only `Value::vector...`

Reply:
I'm not surprised. `vector_for_testing_only` is the only API that allows you to create a vector of complex values. I marked it as "for_testing_only" because it's not checking the layouts of the values, making it possible for one to create invalid values.

I think for now it's okay to use it, considering the natives you are adding are only run locally. However we really should come up with a proper API that offers this functionality. In fact, let me see if I can come up with one.

Reply:
Okay, that layout check can be a bit hard to write. Let's just use `vector_for_testing_only` for now.

Reply:
addressed